tests/integration/test_scheduler/test.py

```diff
@@ -999,3 +999,115 @@ def test_throw_on_unknown_workload():
         assert False, "Exception have to be thrown"
     except Exception as ex:
         assert "RESOURCE_ACCESS_DENIED" in str(ex)
+
+
+def test_config_based_workloads_and_resources():
+    """Test config-based workloads and resources loading and precedence over SQL."""
+    # First, ensure we start clean
+    node.query("system reload config")
+    
+    # Create a test configuration with resources and workloads
+    workloads_config = """
+    <clickhouse>
+        <resources_and_workloads>
+            RESOURCE config_io_read (READ DISK s3_no_resource);
+            RESOURCE config_io_write (WRITE DISK s3_no_resource);
+            WORKLOAD config_all SETTINGS max_bytes_inflight = 1000000 FOR config_io_read, max_bytes_inflight = 2000000 FOR config_io_write;
+            WORKLOAD config_production IN config_all SETTINGS priority = 1, weight = 3
+        </resources_and_workloads>
+    </clickhouse>
+    """
+    
+    # Update config and reload
+    with open("/etc/clickhouse-server/config.d/test_config_workloads.xml", "w") as f:
+        f.write(workloads_config)
+    
+    node.query("system reload config")
+    
+    # Verify config-based entities are loaded
+    resources_result = node.query("SELECT name FROM system.resources WHERE name LIKE 'config_%' ORDER BY name")
+    assert "config_io_read\nconfig_io_write\n" == resources_result
+    
+    workloads_result = node.query("SELECT name FROM system.workloads WHERE name LIKE 'config_%' ORDER BY name")
+    assert "config_all\nconfig_production\n" == workloads_result
+    
+    # Verify scheduler nodes exist for config entities
+    scheduler_result = node.query(
+        "SELECT count() FROM system.scheduler WHERE resource IN ('config_io_read', 'config_io_write') AND path LIKE '%config_%'"
+    )
+    assert int(scheduler_result.strip()) > 0
+    
+    # Try to create SQL entities with same names - should fail
+    try:
+        node.query("CREATE RESOURCE config_io_read (READ DISK s3_no_resource)")
+        assert False, "Should not be able to create resource with same name as config entity"
+    except Exception as ex:
+        assert "defined in configuration" in str(ex)
+    
+    try:
+        node.query("CREATE WORKLOAD config_all")
+        assert False, "Should not be able to create workload with same name as config entity"
+    except Exception as ex:
+        assert "defined in configuration" in str(ex)
+    
+    # Create SQL entities with different names
+    node.query("CREATE RESOURCE sql_io_read (READ DISK s3_no_resource)")
+    node.query("CREATE WORKLOAD sql_all SETTINGS max_bytes_inflight = 500000 FOR sql_io_read")
```

Comment:
It is not allowed to create two root workloads and we already have config_all. Create sql_all as a child instead. Use `sql_all IN config_all` SQL clause for this.